sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py

```diff
@@ -0,0 +1,777 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+# pylint: disable=protected-access
+
+import re
+from collections import namedtuple
+from enum import Enum
+from ._helpers import get_field_scalar_value
+
+
+PageRange = namedtuple("PageRange", "first_page last_page")
+Point = namedtuple("Point", "x y")
+
+
+def get_elements(field, ocr_result):
+    elements = []
+    try:
+        for item in field.elements:
+            extracted_word = None
+            nums = [int(s) for s in re.findall(r'\d+', item)]
+            read = nums[0]
+            line = nums[1]
+            if len(nums) == 3:
+                word = nums[2]
+                ocr_word = ocr_result[read].lines[line].words[word]
+                extracted_word = ExtractedWord._from_generated(ocr_word, page=read+1)
+            ocr_line = ocr_result[read].lines[line]
+            extracted_line = ExtractedLine._from_generated(ocr_line, page=read+1)
+
+            if extracted_word:
+                elements.append(extracted_word)
+            else:
+                elements.append(extracted_line)
+        return elements
+    except TypeError:
+        return None
+
+
+class LengthUnit(str, Enum):
+    """The unit used by the width, height and bounding box properties. For images, the unit is "pixel".
+    For PDF, the unit is "inch".
+    """
+
+    pixel = "pixel"
+    inch = "inch"
+
+
+class TrainStatus(str, Enum):
+    """Status of the training operation.
+    """
+    succeeded = "succeeded"
+    partially_succeeded = "partiallySucceeded"
+    failed = "failed"
+
+
+class ModelStatus(str, Enum):
+    """Status of the model.
+    """
+    creating = "creating"
+    ready = "ready"
+    invalid = "invalid"
+
+
+class BoundingBox:
```

Comment:
I would prefer this to be a named tuple (same names)

Reply:
So a namedtuple with attributes that are namedtuples (Points), just to be clear? Happy to do this, but having trouble figuring out how to document. Seems python 2 doesn't support \__doc__.

Reply:
I'll rephrase, I'd prefer that it has a tuple-like interface. Which means that you can have read-only properties. It doesn't have to be an actual `collections.namedtuple`.

Reply:
done